File: frontend/src/types/Plan.ts
export type Plan = {
    planID: string;
    data: {
        name: string;
        description: string;
        color: string;
        completed: string;
    };
};

type EncryptSpec = {
    [K in keyof Plan["data"]]: boolean;
};

export const PlanEncryptSpec: EncryptSpec = {
    name: true,
    description: true,
    color: false,
    completed: true
};

const newPlan: Plan = {
    planID: '',
    data: {
        name: '',
        description: '',
        color: 'rgb(255, 255, 255)',
        completed: '',
    },
} as const;

export const getNewPlan = (): Plan => structuredClone(newPlan);

File: backend/src/main/java/com/blck/planner/userData/Plan.java
package com.blck.planner.userData;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.util.UUID;

@Table("user_plans")
public record Plan(
        @Id
        @JsonProperty("itemID")
        UUID itemID,
        String userID,
        String name,
        String description,
        String color,
        String completed
) {
    public PlanDTO toDTO() {
        var data = new PlanDTO.Data(name, description, color, completed);
        return new PlanDTO(itemID, data);
    }
}